subprojects/file-collections/src/main/java/org/gradle/api/internal/file/FileCollectionStructureVisitor.java

```diff
@@ -71,11 +71,6 @@ default VisitType prepareForVisit(FileCollectionInternal.Source source) {
      */
     void visitCollection(FileCollectionInternal.Source source, Iterable<File> contents);
 
-    /**
-     * Visits a file tree whose content is generated from some opaque source.
-     */
-    void visitGenericFileTree(FileTreeInternal fileTree, FileSystemMirroringFileTree sourceTree);
-
     /**
      * Visits a file tree at a root file on the file system (potentially filtered).
      */
```

Comment:
This method will now only receive `DirectoryFileTree`s embedded in a `FileTreeAdapter`. I wonder if we could clarify that here. One option is to call this `visitDirectoryFileTree()`. The other is to actually pass the `DirectoryFileTree` instance as a parameter, though I'm not sure if that'd help.

Reply:
I don't think it is important that it is important what is passed in as type for the last argument. The data has already been extracted in the other two arguments, so I think `FileTreeInternal` is perfectly fine here.